tests/test_octodns_provider_cloudflare_processor_ttl.py

```diff
@@ -0,0 +1,42 @@
+from unittest import TestCase
+
+from octodns.record import Record
+from octodns.zone import Zone
+
+from octodns_cloudflare.processor.ttl import TtlToProxy
+
+
+class TestTtlToProxy(TestCase):
+    def test_ttl_to_proxy(self):
+        processor = TtlToProxy('test', ttl=0)
+
+        zone = Zone('unit.tests.', [])
+        zone_expected = Zone('unit.tests.', [])
+
+        with_ttl = Record.new(
+            zone, 'good', {'type': 'A', 'ttl': 0, 'value': '1.2.3.4'}
+        )
+        without_ttl = Record.new(
+            zone, 'bad', {'type': 'A', 'ttl': 10, 'value': '1.2.3.4'}
+        )
+        zone.add_record(with_ttl)
+        zone.add_record(without_ttl)
+
+        expected_with = Record.new(
+            zone,
+            'good',
+            {
+                'type': 'A',
+                'ttl': 0,
+                'value': '1.2.3.4',
+                '_octodns': {'cloudflare': {'proxied': True, 'auto-ttl': True}},
+            },
+        )
+        expected_without = Record.new(
+            zone, 'bad', {'type': 'A', 'ttl': 10, 'value': '1.2.3.4'}
+        )
+        zone_expected.add_record(expected_with)
+        zone_expected.add_record(expected_without)
+
+        added_proxy = processor.process_source_zone(zone)
+        self.assertEqual(zone_expected.records, added_proxy.records)
```

Comment:
Unfortunately due to some implementation details of octoDNS records this doesn't actually compare all the fields of the record, just the names & types. 

As is I believe it would fail b/c the `expected_with` record has a TTL of `0` and the ttl to proxy code sets it to `1`.

Untested/just wrote it here, but maybe something like:

```suggestion
        good = next(r for r in added_proxy.records if r.name == 'good')
        self.assertEqual(1, good.ttl)
        self.assertEqual({'cloudflare': {'proxied': True, 'auto-ttl': True}}, good._octodns)
        bad = next(r for r in added_proxy.records if r.name == 'bad')
        self.assertEqual(10, bad.ttl)
        self.assertFalse('cloudflare' in bad._octodns)
```